### PR TITLE
storage: allow all storage collections to be written to a Sink

### PR DESCRIPTION
**Pull request by aljoscha**
Notably, this now allows creating a Sink that has a Table as input.

This removes the check for transitive dependencies when `reuse_topic =
true` and also removes the (already dead before this) code that had to
do with transitive dependencies of sinks. This code was used before
Sinks where moved to run in a `storaged` and to work only off of storage
collections.

Closes https://github.com/MaterializeInc/materialize/issues/14200

### Tips for reviewer

@philip-stoev Do we want more tests for that, or are the re-enabled ones fine for now and you are anyways working on more tests that use this already?

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Review by philip-stoev — APPROVED** [review]
I have more tests sitting in a branch (that confirm that table-based sinks work OK when enabled), so please merge this on green and I will then merge my tests with a separate PR. Thanks!

File: src/storage/src/types/sinks.proto [hunk 1]
```diff
@@ -85,7 +85,6 @@ message ProtoKafkaSinkConnection {
     optional ProtoPublishedSchemaInfo published_schema_info = 7;
     ProtoKafkaSinkConsistencyConnection consistency = 8;
     bool exactly_once = 9;
-    repeated mz_repr.global_id.ProtoGlobalId transitive_source_dependencies = 10;
```
Comment (antiguru) [inline src/storage/src/types/sinks.proto:88]: Mark tag 10 as reserved for backwards-compatibility.

**Comment by aljoscha** [comment]
@philip-stoev I had to change `table-persistence-*-sinks.td` to explicitly use `reuse_topic = true`. That intuitively makes sense and I think theses tests only passed before Chae's changes that always sink from a storage collection because we would always re-create the topic and write out everything. With the new changes and `reuse_topic = true` we seem to be doing the _right thing_. What do you think?

**Comment by philip-stoev** [comment]
Yes totally, my mistake to code the test like that originally. Thanks a lot!

**Pull request merged**